exporter/opentelemetry-exporter-prometheus-remote-write/tests/test_prometheus_remote_write_exporter.py

```diff
@@ -0,0 +1,337 @@
+        with self.assertRaises(ValueError):
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint",
+                basic_auth={"username": "test_username"},
+            )
+
+    def test_invalid_conflicting_passwords_param(self):
+        with self.assertRaises(ValueError):
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint",
+                basic_auth={
+                    "username": "test_username",
+                    "password": "test_password",
+                    "password_file": "test_file",
+                },
+            )
+
+    def test_invalid_timeout_param(self):
+        with self.assertRaises(ValueError):
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint", timeout=0
+            )
+
+    def test_valid_tls_config_param(self):
+        tls_config = {
+            "ca_file": "test_ca_file",
+            "cert_file": "test_cert_file",
+            "key_file": "test_key_file",
+            "insecure_skip_verify": True,
+        }
+        exporter = PrometheusRemoteWriteMetricsExporter(
+            endpoint="/prom/test_endpoint", tls_config=tls_config
+        )
+        self.assertEqual(exporter.tls_config["ca_file"], tls_config["ca_file"])
+        self.assertEqual(
+            exporter.tls_config["cert_file"], tls_config["cert_file"]
+        )
+        self.assertEqual(
+            exporter.tls_config["key_file"], tls_config["key_file"]
+        )
+        self.assertEqual(
+            exporter.tls_config["insecure_skip_verify"],
+            tls_config["insecure_skip_verify"],
+        )
+
+    # if cert_file is provided, then key_file must also be provided
+    def test_invalid_tls_config_cert_only_param(self):
+        tls_config = {"cert_file": "value"}
+        with self.assertRaises(ValueError):
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint", tls_config=tls_config
+            )
+
+    # if cert_file is provided, then key_file must also be provided
+    def test_invalid_tls_config_key_only_param(self):
+        tls_config = {"cert_file": "value"}
+        with self.assertRaises(ValueError):
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint", tls_config=tls_config
+            )
+
+
+class TestConversion(unittest.TestCase):
+    # Initializes test data that is reused across tests
+    def setUp(self):
+        self._test_metric = Counter(
+            "testname", "testdesc", "testunit", int, None
+        )
+        self._exporter = PrometheusRemoteWriteMetricsExporter(
+            endpoint="/prom/test_endpoint"
+        )
+
+        def generate_record(aggregator_type):
+            return ExportRecord(
+                self._test_metric, None, aggregator_type(), Resource({}),
+            )
+
+        self._generate_record = generate_record
```

Comment:
It felt repetitive to use the MetricRecord contructor every time since most of the arguments were the same in each test, so I thought I could just create a method that creates a test_record with the required aggregator type.

Reply:
Ok, but why define a private attribute for `generate_record`? Also, if you find yourself repeating arguments in several calls, then [`partial`](https://docs.python.org/3/library/functools.html#functools.partial) can help.

Reply:
I decided to just use the constructor despite some repeating arguments in each method to make it clear what the properties of the test record are for someone just looking at one test.

Side Note: You are right, there is no reason for that test helper method to be private. My understanding of what should be public/private in the test class was incorrect.